### ISSUE TITLE
New release 2.4.0

### ISSUE THREAD

**Issue by srbaker** [TracksApp/tracks#2218]
Is there a "punch list" somewhere for the next release? Basically a list of all of the things that need to be done in order for a release to be made? Perhaps a release tag in issues? I'd like to help punch things out.

I know updated screenshots for the website are one of those things, shall I start there?

**Comment by ZeiP** [comment]
There's an release issue #2029 which points to #1968. Additionally there's the resurrection project board, which has some tasks: https://github.com/TracksApp/tracks/projects/1

**Comment by mattr-** [comment]
I just closed #2029 because I removed all the feature files so that's no longer relevant. I appreciate you pointing that out.

I haven't really focused on making a next release. I've got a lot of code cleanup and UI ideas that I'd like to play and experiment with. The transition to bootstrap based UI elements hasn't really gone well and the UI really shows it's age in spots. 

In short, it sounds like I need to start writing up more issues to get the things that are in my head out, and allow other folks to pick up things and work on them. I'll spend some time doing that this week.

**Comment by srbaker** [comment]
@mattr- I would certainly appreciate that. I have a lot of inclination to work on things right now, so a brain dump would be appreciated. Better yet, if there are several of us, perhaps a call?

**Comment by ZeiP** [comment]
We should think about a new release, at least a minor one (2.3.1). The last release has been four years ago, so basically it seems like to project isn't live any more – which isn't true. Additionally we have a wealth of fixes to release:

- Fixed numerous XML API endpoints and their token authentication
- Fixed smaller bugs in the user interface
- Upgraded a ton of dependencies
- Fixed setting the default context and tags based on project (once #2238 is merged)
- Changed stats charts to use Javascript (once #2231 is merged)

Especially the charts changed could even warrant a 2.4 release, IMO.

I'd suggest making a release after at least PR's #2238 and #2231 are merged. Perhaps we should also update the web page per the Resurrection project board. Any other issues you think are necessary for a release?

**Comment by mattr-** [comment]
The next release will be 2.4. There's too many changes for a patch release.

The biggest blocker for release is that we can't publish it to the website. The publishing workflow for TracksApp/tracksapp.github.io needs to be updated so we can actually publish changes to the website and then we can cut all the releases we want.

**Comment by ZeiP** [comment]
Ok. What's the issue with the website? I actually took a peek at it today when I noticed that some of the resources aren't through HTTPS. I was wondering why it isn't done using Github Pages, is that what you're planning? I'd be happy to help migrate the current content if necessary.

**Comment by mattr-** [comment]
It's an older Octopress based site, and the setup is rather tempermental. If you want to port it to plain Jekyll, that'd be fine with me. That's what I was planning on doing.

**Comment by ZeiP** [comment]
Ok, I'll take a peek at it.

**Comment by ZeiP** [comment]
There's a PR for the change at https://github.com/TracksApp/tracksapp.github.io/pull/26.

**Comment by ZeiP** [comment]
The web site is working again. There seems to be a milestone list in https://github.com/TracksApp/tracks/milestone/23 – perhaps we should check the issues in that list and check which of those actually need to be done for 2.4.0? That would work as a useful punch list for the release.

**Comment by mattr-** [comment]
I want to move everything out of that milestone. I don't really want to do much more with 2.4.0 than we've already done. The rails upgrade and the chart changes and any other bugs that get found are going to be enough, I think.

**Comment by ZeiP** [comment]
I mostly agree. There are three open issues that probably need a major release once fixed, I wonder if we could fix those in this release also: https://github.com/TracksApp/tracks/milestone/23

**Comment by mattr-** [comment]
:thumbsup: Those should definitely be addressed before 2.4

**Comment by C-Otto** [comment]
I think a fix for #2242 should be part of the release, it's very annoying. Thanks for moving this project forward :)

**Comment by ZeiP** [comment]
I'd say that we have a release ready after merging the three PR's I created today. #2247 was also added to the milestone, but I wouldn't consider it that urgent – of course if someone has time to figure it out, all the better.

**Comment by ZeiP** [comment]
I suggest we make a release out of it. The 2.4.0 project board contains #2259 and #2247, but I think both of them are small enough to do in a minor version and not important enough to stall releasing.

**Comment by ZeiP** [comment]
@mattr-, what do you think about making the release?

**Comment by mattr-** [comment]
Sounds good. I'll start working on it. Setting up an upgrade system so that I can test out the instructions and everything, and then i'll get to publishing a release.

**Comment by ZeiP** [comment]
#2277 blocks this for now.

**Comment by ZeiP** [comment]
@mattr-, is there something I could do to help with the release?

**Comment by srbaker** [comment]
❤️

**Comment by mattr-** [comment]
There is now a `v2.4.0` tag on the repo. I'll look at other odds and ends and write a release announcement by EOD tomorrow.

**Comment by ZeiP** [comment]
@mattr-, have you gotten around to making the Web page changes? I can make those too, if you haven't had the time.

**Comment by ZeiP** [comment]
@mattr-, I made a PR of the release Web site changes, want to check them? https://github.com/TracksApp/tracksapp.github.io/pull/30

**Comment by ZeiP** [comment]
Release news has been published, I think the release is done now.
